scripts/test_adv_IM.py

```diff
@@ -14,23 +24,81 @@
         "adv_im_model", nargs="+", type=str, help="list of adv_IM models that ran"
     )
 
+    parser.add_argument(
+        "--components",
+        default=[const.Components.c000.str_value, const.Components.c090.str_value],
+        nargs="+",
+        choices=list(const.Components.iterate_str_values()),
+        help="list of component that ran",
+    )
+
     args = parser.parse_args()
 
+    return args
+
+
+def main(sim_dir, adv_im_model, components, exit=True):
+
     failed = []
-    for model in args.adv_im_model:
-        csv_path = os.path.join(args.sim_dir, "IM_calc/{}.csv".format(model))
+
+    for model in adv_im_model:
+        # check if csv exist
+        csv_path = os.path.join(sim_dir, "IM_calc/{}.csv".format(model))
+        im_dir = get_im_calc_dir(sim_dir)
         # check file exist first to prevent crash
         try:
             df = pd.read_csv(csv_path)
         except FileNotFoundError:
             failed.append(model)
             print("csv for {} does not exist".format(model))
         else:
+            # check for null/nan
             if df.isnull().values.any():
                 failed.append(model)
+
+            # if csv exist:
+            # check station count, if match, skip rest of checks
+            csv_station_count = len(df.station.unique())
+            # glob for station folders
+            list_folders = [x for x in os.listdir(im_dir) if os.path.isdir(x)]
+            if csv_station_count == len(list_folders):
+                # station folder count matches csv.station.count
+                continue
+            else:
+                for station_dir in list_folders:
+                    # check if folder is a station run folder
+                    station_model_dir = os.path.join(station_dir, model)
+                    station_name = os.path.basename(station_dir)
+                    if not os.path.isdir(station_model_dir) or (
+                        station_name in df.station.unique()
+                    ):
+                        # does not match naming for a station run, possibly created by something else, skipping
+                        continue
+                    else:
```

Comment:
Again, negate and drop continue or drop the else and de-indent

Reply:
wouldn't it be clearer to someone that reads the code to know what test these lines are doing what which test has higher priority by having the if, else and indent?
As these lines has no practical impact on the script nor compatibility issues, i dont see it as an issue? 
the computation time should be exactly the same as well.

Reply:
If you drop the else and de-indent then the intent will be clear, because anyone reading it will know the effect continue has
Negating the test is probably less clear as it's harder to track down where the loop ends.
I'm still in favour of dropping the else and indentation level though

Reply:
In general computation time isn't a factor. It's more about being able to clearly see the progression / layout of the code.